CosmicReco/src/PDFFit.cc

```diff
@@ -299,27 +299,31 @@ double FullDriftFit::operator()(const std::vector<double>& x) const {
 }
 
 double GaussianDriftFit::operator()(const std::vector<double>& x) const {
-  double a0 = x[0];
-  double b0 = x[1];
-  double a1 = x[2];
-  double b1 = x[3];
-  double t0 = x[4];
+  double const&a0 = x[0];
```

Comment:
Is there a reason these are const& instead of local variables?  conventional wisdom is that const& doesn't save on performance for fundamental types, but maybe that's not true here.

Reply:
Since the function is called many times I'd assumed it couldn't hurt to refer to the vector elements / track params by reference, I may be wrong

Reply:
For unoptimized code the references should be small speed improvement over local variables, since references are effectively compile time aliases while local variables require stack operations and copies.   For -O3 I would be surprised if there were any runtime speed difference between the two choices.

I do like the style of converting the elements of the std::vector into well named variables; for me, at least, it makes the downstream code easier to read and that's valuable.

It's probably too late now but what drove the choice between std::vector vs std::array vs a small struct with elements a0, b0, a1, b1, t0?

Reply:
Hi Rob,

Minuit deals with parameters in `std::vector` form while doing the fit, so the `operator()` function can't change. 

Within my namespace `AlignmentUtilities` I did make a struct to make it easier for me to pass around the fit parameters between functions. It also removed a lot of boilerplate constructing intercept and direction `Hep3Vectors` from those parameters everywhere.

Reply:
The original std::vector is just the inherited interface from Minuit